File: src/Clients/accountant/src/models/entities/automaticTransaction.ts
import { Syncable } from "models/syncable";

export class AutomaticTransaction implements Syncable {
  categoryName: string;
  synced = false;

  constructor(
    public id: number,
    public isDeposit: boolean,
    public categoryId: number,
    public amount: number,
    public currency: string,
    public description: string,
    public dayInMonth: number,
    public createdDate: Date,
    public modifiedDate: Date
  ) {}
}

File: src/Background/Jobs/Models/Accountant/AutomaticTransaction.cs
namespace Jobs.Models.Accountant;

internal class AutomaticTransaction
{
    internal int UserId { get; set; }
    internal bool IsDeposit { get; set; }
    internal int? CategoryId { get; set; }
    internal decimal Amount { get; set; }
    internal string Currency { get; set; }
    internal string Description { get; set; }
}
